### PR TITLE
Enable TypeScript's strict mode and fix failing tests

### PR DESCRIPTION
**Pull request by adamkobor**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] API docs have been updated (by executing `npm run documentation`)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Other: The changes are making able to compile the SDK with `"strict": true` option in TypeScript, and the already broken tests also has been fixed. 

## What is the current behavior?
TypeScript compilation fails if we try to use `"strict": true`, and tests are braking (#140)

## What is the new behavior?
The SDK is able to compile with strict mode and failing tests has been fixed.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by rijkvanzanten** [comment]
@jooola any thoughts? I know you've been looking at optimizing this. Just wanna make sure we're not causing any conflicts 🙂

File: test/configuration.spec.ts [hunk 1]
```diff
@@ -141,7 +135,7 @@ describe("Configuration", () => {
         );
 
         expect(config.url).to.equal("https://demo-api.getdirectus.com");
-        expect(config.project).to.equal("_"); // default value
+        expect(config.project).to.be.undefined; // default value
```
Comment (jooola) [inline test/configuration.spec.ts]: Comment should be erased I think

File: test/configuration.spec.ts [hunk 2]
```diff
@@ -141,7 +135,7 @@ describe("Configuration", () => {
...
         expect(config.token).to.be.undefined;
         expect(config.localExp).to.be.undefined;
         expect(config.tokenExpirationTime).to.equal(30000); // default value
```
Comment (jooola) [inline test/configuration.spec.ts]: Same as above

File: tsconfig.test.json [hunk 3]
```diff
@@ -1,6 +1,8 @@
 {
   "extends": "./tsconfig.json",
-  "compilerOptions": {},
+  "compilerOptions": {
```
Comment (jooola) [inline tsconfig.test.json]: No need since we extend the first tsconfig.json

File: tsconfig.meta.json [hunk 4]
```diff
@@ -1,6 +1,7 @@
 {
   "extends": "./tsconfig.json",
   "compilerOptions": {
+    "strict": true,
```
Comment (jooola) [inline tsconfig.meta.json]: No need since we extend the first tsconfig.json

**Comment by rijkvanzanten** [comment]
@adamkobor mind resolving those tiny tweaks? 🙂 I know some of the feedback points from @jooola were already like that before, but this is a good time to make sure everything is 💯

**Comment by adamkobor** [comment]
Sure, I will take care of it tomorrow 👌

File: test/configuration.spec.ts (hunk 1 above)
Comment (adamkobor) [reply test/configuration.spec.ts]: That's already fixed 👌

File: test/configuration.spec.ts (hunk 2 above)
Comment (adamkobor) [reply test/configuration.spec.ts]: Same here

**Comment by adamkobor** [comment]
@rijkvanzanten 
I've resolved the issues above, however I'm facing a weird issue, when I try to rerun `npm run documentation`.  Typedoc removes important pieces from the existing docs, I think it's reproducible, so you can check it too if you run the command.

**Comment by rijkvanzanten** [comment]
I think we'll have to spend a separate PR on either integrating Typedoc properly, of getting rid of it completely. Lets not consider it in this one though 🙂

**Comment by adamkobor** [comment]
Hey @rijkvanzanten, do you have any estimation about the release of this? 🙏

**Comment by rijkvanzanten** [comment]
Just released it 👍 It's under 6.0.3.

@jooola GH action for release worked like a charm 👍

**Comment by jooola** [comment]
@rijkvanzanten Glad you like it, I'll push another PR. A few things needs to be improved still.

**Pull request merged**